cardboardlint/linter_import.py

```diff
@@ -0,0 +1,56 @@
+"""Linter for import conventions.
+
+This script counts the number of bad imports. The following is not allowed in a package:
+
+* Importing from the top-level of its own package, e.g.:
+
+  .. code-block:: python
+
+        from package import foo
+"""
+import os
+import codecs
+from cardboardlint.common import Message, get_filenames
+
+
+def linter_import(config, files_lines):
+    """Linter for checking import statements.
+
+    Parameters
+    ----------
+    config : dict
+        Dictionary that contains the configuration for the linter
+    files_lines : dict
+        Dictionary of filename to the set of line numbers (that have been modified).
+        See `run_diff` method in `carboardlint`
+    """
+    messages = []
+
+    # Find all (sub)package names, from which one should not import directly
+    packages = []
+    filenames = get_filenames(config['directories'], config['include'], config['exclude'],
+                              files_lines=files_lines)
+    # NOTE: does this include the top module's __init__.py? e.g. 'from cardboardlint import'
+    # FIXME: assumes that each filename starts from only the module's directory (without ./)
```

Comment:
I'm not sure if I understand correctly. `get_filenames` should not put a leading `./` in front of the filenames. Can that happen?

The loop below assumes that the source tree contains a `package` or `package/subpackage` (etc.) directory for every Python (sub)package in the project. This convention is not set in stone but we tend to stick to it. It is something we may always generalize if needed.